staging/src/k8s.io/apiserver/pkg/server/options/server_run_options.go

```diff
@@ -183,6 +188,53 @@ func validateHSTSDirectives(hstsDirectives []string) error {
 	return errors.NewAggregate(allErrors)
 }
 
+func validateCorsAllowedOriginList(corsAllowedOriginList []string) error {
+	allErrors := []error{}
+	validateRegexFn := func(regexpStr string) error {
+		if _, err := regexp.Compile(regexpStr); err != nil {
+			return fmt.Errorf("--cors-allowed-origins has an invalid regular expression: %v", err)
+		}
+
+		// the regular expression should pin to the start and end of the host
+		// in the origin header, this will prevent CVE-2022-1996.
+		// possible ways it can pin to the start of host in the origin header:
+		//   - match the start of the origin with '^'
+		//   - match what separates the scheme and host with '//' or '://',
+		//     this pins to the start of host in the origin header.
+		// possible ways it can match the end of the host in the origin header:
+		//   - match the end of the origin with '$'
+		//   - with a capture group that matches the host and port separator '(:|$)'
+		var pinStart, pinEnd bool
+		for _, prefix := range []string{"^", "://", "//"} {
+			if strings.HasPrefix(regexpStr, prefix) {
+				pinStart = true
+				break
+			}
+		}
+		for _, suffix := range []string{"$", "$)"} {
+			if strings.HasSuffix(regexpStr, suffix) {
```

Comment:
As long as they are using `$` or `:` anywhere in their expression, they are considering matching the end of the hostname. Relax this to just look at containing either of those:

```suggestion
		for _, suffix := range []string{"$", ":"} {
			if strings.Contains(regexpStr, suffix) {
```

Reply:
I have relaxed the validation to `Contains` - `:` is also in the scheme